arangod/Aql/AqlValue.cpp

```diff
@@ -376,7 +422,12 @@ AqlValue AqlValue::getIdAttribute(CollectionNameResolver const& resolver,
     case VPACK_INLINE:
     case VPACK_MANAGED_SLICE:
     case VPACK_MANAGED_STRING:
+    case VPACK_SUPERVISED_SLICE:
       if (auto s = slice(t); s.isObject()) {
+        ResourceMonitor* rm = nullptr;
+        if (this->type() == VPACK_SUPERVISED_SLICE) {
+          rm = this->_data.supervisedSliceMeta.getResourceMonitor();
+        }
```

Comment:
This pattern is repeated several times - can we just add a function `getResourceMonitor`?
```
ResourceMonitor* getResourceMonitor() {
  if (type() == VPACK_SUPERVISED_SLICE) {
    return _data.supervisedSliceMeta.getResourceMonitor();
  }
  return nullptr;
}
```